### ISSUE TITLE
[Beta] Issues related to 404 - Page not found

### ISSUE THREAD

**Issue by vkWeb** [freeCodeCamp/freeCodeCamp#16949]
### Issue Description
- [ ] The 404 error is visible only for a few milliseconds, to reproduce go to https://beta.freecodecamp.org/en/challenges/sub/invalid
**Note:** We have three different 404 ([first](https://beta.freecodecamp.org/en/invalidURL), [second](https://beta.freecodecamp.org/en/challenges/parent/invalidURL) & [third](https://beta.freecodecamp.org/en/challenges/parent/child/sub/invalidURL)). This is only applicable to the second type of 404.

- [ ] The `we could not found...` warning flash is not dismissing when trying to close. This is only applicable to the [third type](https://beta.freecodecamp.org/en/challenges/parent/child/sub/invalidURL) of 404. 

### Browser Information
* Browser Name, Version: Chrome(65.0.3325.181) and Firefox(59.0.1).
* Operating System: Windows 10 Pro 32 Bit.
* Mobile, Desktop, or Tablet: Desktop.

**Comment by anhtran1906** [comment]
I found the issue. Basically in the main redux file, challengeSelector should return {} when challengeName is not found in challengeMap (when accessing invalid challenge) instead of jumping to firstSelector and returns the state there. 
<img width="759" alt="screen shot 2018-04-03 at 11 50 01 pm" src="https://user-images.githubusercontent.com/20694114/38287808-c671fdcc-379a-11e8-8cf9-f24c7e5b9223.png">
I added my code and it worked: 
![workingversion](https://user-images.githubusercontent.com/20694114/38287801-b5ba3756-379a-11e8-8a83-d11967a42b47.gif)
Will submit a PR now.
